kedro/framework/cli/micropkg.py

```diff
@@ -390,12 +390,14 @@ def _is_within_directory(directory: Path, target: Path) -> bool:
 
 
 def safe_extract(tar: tarfile.TarFile, path: Path) -> None:
+    safe_members = []
     for member in tar.getmembers():
         member_path = path / member.name
         if not _is_within_directory(path, member_path):
             # noqa: broad-exception-raised
             raise Exception("Failed to safely extract tar file.")
-    tar.extractall(path)  # nosec B202
+        safe_members.append(member)
+    tar.extractall(path, members=safe_members)  # nosec B202
```

Comment:
> The [B202:tarfile_unsafe_members](https://bandit.readthedocs.io/en/latest/plugins/b202_tarfile_unsafe_members.html) documentation says to pass a callable as the members argument but that’s not supported in [the official type signature](https://github.com/python/typeshed/blob/1d7f0d087b36d07ce34edca5ab119fdaabf7aed7/stdlib/tarfile.pyi#L295) and not implemented in [CPython stdlib](https://github.com/python/cpython/blob/025995feadaeebeef5d808f2564f0fd65b704ea5/Lib/tarfile.py#L2254C5-L2254C5).

https://github.com/PyCQA/bandit/issues/1038

Please add a link to bandit#1038 for future clarity

Reply:
done!